addons/mass_mailing_sms/views/mailing_mailing_views.xml

```diff
@@ -98,12 +98,18 @@
                 <attribute name="attrs">{'invisible': ['|',('mailing_type', '!=', 'mail'),('state', 'in', ('draft','test'))]}</attribute>
             </xpath>
             <!-- Form -->
+            <xpath expr="//field[@name='subject']" position="attributes">
+                <attribute name="attrs">{'invisible': [('mailing_type', '!=', 'mail')], 'required': [('mailing_type', '=', 'mail')]}</attribute>
```

Comment:
You also have to re-specify the readonly attribute since this will override the whole attrs.